bash on windows has error

I use glide in bash on windows. when i exec `glide install`.i got the follow error message
```shell
[INFO]  --> Exporting golang.org/x/net
[INFO]  --> Exporting golang.org/x/crypto/acme/autocert
[INFO]  --> Exporting golang.org/x/crypto
[INFO]  Replacing existing vendor dependencies
[ERROR] remove /tmp/glide-vendor220918663/vendor/github.com/google/go-github/github: directory not empty
[ERROR] Unable to export dependencies to vendor directory: remove /mnt/c/Users/Administrator/go/src/github.com/freshcn/go-id-builder/vendor/github.com/google/go-github/github: directory not empty
```
there has the same error on  v0.12.3 and 0.13.0-dev

seeing this also

```bash
$ glide -v
glide version 0.13.0-dev
```

with error

```
[ERROR] remove /tmp/glide-vendor361549384/vendor/google.golang.org/api: directory not empty                                          
[ERROR] Unable to export dependencies to vendor directory: remove /root/gopath/src/github.com/xx/yy/vendor/google.golang.org/api: directory not empty
```

@djensen47 I tried clearing cache and running glide update again. Still the same error.

> [ERROR] Unable to export dependencies to vendor directory: remove C:\Workspace\...\vendor\github.com\chzyer\readline\example: The directory is not empty.

```
$ git clone git@github.com:Masterminds/glide.git $GOPATH/github.com/Masterminds/glide
$ cd $GOPATH/github.com/Masterminds/glide
$ git checkout -b windows-many-files origin/fix/windows-many-files
$ make build
$ mv glide $GOPATH/bin
```

Go to your project and use the new glide but be sure that you're using the new one...
```
$ which glide
```

Next, I cleared everything away.
```
$ rm -rf vendor
$ glide cc
```

@mattfarina My first attempt at `glide cc` failed with this error:
```
[ERROR] Unable to clear the cache: remove .glide/cache/src/https-github.com-fatih-color/vendor/golang.org/x/sys/unix: directory not empty
```
I tried again and it worked the second time.

The moment of truth ...
```
$ glide update
```
🥁  (to be continued)

Closer...

make build
go build -o glide -ldflags "-X main.version=v0.12.1-81-ga7ffb17" glide.go
glide.go:24:2: cannot find package "github.com/Masterminds/glide/action" in any of:
        /usr/lib/go-1.6/src/github.com/Masterminds/glide/action (from $GOROOT)
        /root/marcb/Code/go/src/github.com/Masterminds/glide/action (from $GOPATH)
glide.go:25:2: cannot find package "github.com/Masterminds/glide/cache" in any of:
        /usr/lib/go-1.6/src/github.com/Masterminds/glide/cache (from $GOROOT)
        /root/marcb/Code/go/src/github.com/Masterminds/glide/cache (from $GOPATH)
glide.go:26:2: cannot find package "github.com/Masterminds/glide/msg" in any of:
        /usr/lib/go-1.6/src/github.com/Masterminds/glide/msg (from $GOROOT)
        /root/marcb/Code/go/src/github.com/Masterminds/glide/msg (from $GOPATH)
glide.go:27:2: cannot find package "github.com/Masterminds/glide/path" in any of:
        /usr/lib/go-1.6/src/github.com/Masterminds/glide/path (from $GOROOT)
        /root/marcb/Code/go/src/github.com/Masterminds/glide/path (from $GOPATH)
glide.go:28:2: cannot find package "github.com/Masterminds/glide/repo" in any of:
        /usr/lib/go-1.6/src/github.com/Masterminds/glide/repo (from $GOROOT)
        /root/marcb/Code/go/src/github.com/Masterminds/glide/repo (from $GOPATH)
glide.go:29:2: cannot find package "github.com/Masterminds/glide/util" in any of:
        /usr/lib/go-1.6/src/github.com/Masterminds/glide/util (from $GOROOT)
        /root/marcb/Code/go/src/github.com/Masterminds/glide/util (from $GOPATH)
glide.go:31:2: cannot find package "github.com/codegangsta/cli" in any of:
        /usr/lib/go-1.6/src/github.com/codegangsta/cli (from $GOROOT)
        /root/marcb/Code/go/src/github.com/codegangsta/cli (from $GOPATH)
Makefile:8: recipe for target 'build' failed
make: *** [build] Error 1

--- a/path/strip.go
+++ b/path/strip.go
@@ -30,7 +30,7 @@ func StripVendor() error {
 			if _, err := os.Stat(path); err == nil {
 				if info.IsDir() {
 					msg.Info("Removing: %s", path)
-					return os.RemoveAll(path)
+					return CustomRemoveAll(path)
 				}
 
 				msg.Debug("%s is not a directory. Skipping removal", path)

--- a/path/winbug.go
+++ b/path/winbug.go
@@ -1,0 +1,96 @@
+package path
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"bytes"
+	"io/ioutil"
+
+	"github.com/Masterminds/glide/msg"
+)
+
+// This file and its contents are to handle a Windows bug where large sets of
+// files fail when using the `os` package. This has been seen in Windows 10
+// including the Windows Linux Subsystem.
+// Tracking the issue in https://github.com/golang/go/issues/20841. Once the
+// upstream issue is fixed this change can be reverted.
+
+// CustomRemoveAll is similar to os.RemoveAll but deals with the bug outlined
+// at https://github.com/golang/go/issues/20841.
+func CustomRemoveAll(p string) error {
+
+	// Handle the windows case first
+	if runtime.GOOS == "windows" {
+		msg.Debug("Detected Windows. Removing files using windows command")
+		cmd := exec.Command("cmd.exe", "/c", fmt.Sprintf("rd /s /q %s", p))
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("Error removing files: %s. output: %s", err, output)
+		}
+	} else if detectWsl() {
+		cmd := exec.Command("rm", "-rf", p)
+		output, err2 := cmd.CombinedOutput()
+		msg.Debug("Detected Windows Subsystem for Linux. Removing files using subsystem command")
+		if err2 != nil {
+			return fmt.Errorf("Error removing files: %s. output: %s", err2, output)
+		}
+	}
+	return os.RemoveAll(p)
+}
+
+// CustomRename is similar to os.Rename but deals with the bug outlined
+// at https://github.com/golang/go/issues/20841.
+func CustomRename(o, n string) error {
+
+	// Handking windows cases first
+	if runtime.GOOS == "windows" {
+		msg.Debug("Detected Windows. Moving files using windows command")
+		cmd := exec.Command("cmd.exe", "/c", "move", o, n)
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("Error moving files: %s. output: %s", err, output)
+		}
+
+		return nil
+	} else if detectWsl() {
+		cmd := exec.Command("mv", o, n)
+		output, err2 := cmd.CombinedOutput()
+		msg.Debug("Detected Windows Subsystem for Linux. Removing files using subsystem command")
+		if err2 != nil {
+			return fmt.Errorf("Error moving files: %s. output: %s", err2, output)
+		}
+
+		return nil
+	}
+
+	return os.Rename(o, n)
+}
+
+var procIsWin bool
+var procDet bool
+
+func detectWsl() bool {
+
+	if !procDet {
+		procDet = true
+		_, err := os.Stat("/proc/version")
+		if err == nil {
+			b, err := ioutil.ReadFile("/proc/version")
+			if err != nil {
+				msg.Warn("Unable to read /proc/version that was detected. May incorrectly detect WSL")
+				msg.Debug("Windows Subsystem for Linux detection error: %s", err)
+				return false
+			}
+
+			if bytes.Contains(b, []byte("Microsoft")) {
+				msg.Debug("Windows Subsystem for Linux detected")
+				procIsWin = true
+			}
+		}
+	}
+
+	return procIsWin
+}

--- a/repo/installer.go
+++ b/repo/installer.go
@@ -380,12 +380,12 @@ func (i *Installer) Export(conf *cfg.Config) error {
 		}
 	}
 
-	err = os.RemoveAll(i.VendorPath())
+	err = gpath.CustomRemoveAll(i.VendorPath())
 	if err != nil {
 		return err
 	}
 
-	err = os.Rename(vp, i.VendorPath())
+	err = gpath.CustomRename(vp, i.VendorPath())
 	if terr, ok := err.(*os.LinkError); ok {
 		return fixcle(vp, i.VendorPath(), terr)
 	}

